### DISCUSSION TITLE
Capacitor 3: iOS / Swift custom plugin some migration questions

### DISCUSSION THREAD

**Discussion by Nasicus** [ionic-team/capacitor#4924] [Q&A] [answered]
First off: I'm the worst swift programmer in history and don't do a lot there...

But here comes my questions:

-----------------

pre 3.0 I had:

```
        let hints = call.get("hints", [String: String].self);
        let loginHint = hints!["login_hint"];
```
`call.get` does not exist anymore, and I'm too stupid to find out how it works when calling `call.options`.

---------------

pre 3.0 I had:

```
        let webviewParameters = MSALWebviewParameters(parentViewController: self.bridge.viewController);
```
I updated this to:

```
        let webviewParameters = MSALWebviewParameters(parentViewController: (self.bridge?.viewController)!);
```
is it ok to assume that viewController always needs to be there? if this is now "really optional" it would mean that I'd have to change my whole "flow" and return or throw an exception if it's null => should  I acutally do that?

**Comment by jcesarmobile** [comment, marked as answer]
For 1, use `call.getObject("hints")`

For 2, bridge and viewController are now weak, which means they are optional in Swift, but it's up to you if you want to force unwrap or check their existence first.

**Reply by Nasicus** [reply]
Thanks for the help!
